File: backend/app/models/product.py
from uuid import UUID

from sqlalchemy import Boolean, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class Product(Base):
    __tablename__ = "products"

    vendor_id: Mapped[UUID] = mapped_column(
        ForeignKey("users.id"),
        nullable=False,
        index=True,
    )

    name: Mapped[str] = mapped_column(
        String(200),
        nullable=False,
    )

    slug: Mapped[str] = mapped_column(
        String(220),
        nullable=False,
        unique=True,
        index=True,
    )

    description: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    category_id: Mapped[UUID] = mapped_column(
        ForeignKey("categories.id"),
        nullable=False,
        index=True,
    )

    is_active: Mapped[bool] = mapped_column(
        Boolean,
        nullable=False,
        default=True,
    )

File: frontend/src/types/product.ts
export interface Product {
    id: string;
    vendor_id: string;
    name: string;
    slug: string;
    description: string | null;
    category_id: string;
    is_active: boolean;
}

export interface ProductVariant {
    id: string;
    product_id: string;
    sku: string;
    brand: string | null;
    manufacturer: string | null;
    color: string | null;
    size: string | null;
    unit_price: number;
    is_active: boolean;
}